cpp/src/replace/replace.cu

```diff
@@ -77,47 +102,97 @@ namespace{ //anonymous
   /* ----------------------------------------------------------------------------*/
   struct replace_kernel_forwarder {
     template <typename col_type>
-    void operator()(void*       d_col_data,
-                    gdf_size_type      nrows,
-                    const void* d_old_values,
-                    const void* d_new_values,
-                    gdf_size_type      nvalues)
+    void operator()(const gdf_column  &input_col,
+                    const gdf_column &old_values,
+                    const gdf_column &new_values,
+                    gdf_column       &output)
```

Comment:
Adding a `cudaStream_t` arg here will make it easier in the future to add stream support. Then make all the CUDA calls inside this function use the stream arg.